server/util/flagutil/yaml/yaml.go

```diff
@@ -599,6 +611,74 @@ func RemoveEmptyMapsFromYAMLMap(m map[string]any) map[string]any {
 	return m
 }
 
+func expandValue(value string) (string, error) {
+	ctx := context.Background()
+	var expandErr error
+	expandedValue := os.Expand(value, func(s string) string {
+		if strings.HasPrefix(s, externalSecretPrefix) {
+			if SecretProvider == nil {
+				expandErr = status.UnavailableError("config references an external secret but no secret provider is available")
+			} else {
+				name := strings.TrimPrefix(s, externalSecretPrefix)
+				secret, err := SecretProvider.GetSecret(ctx, name)
+				if err != nil {
+					expandErr = status.UnavailableErrorf("could not retrieve config secret %q: %s", name, err)
+				}
+				return string(secret)
+			}
+		}
+		return os.Getenv(s)
+	})
+	if expandErr != nil {
+		return "", expandErr
+	}
+	return expandedValue, nil
+}
+
+func expandSliceValues(yamlSlice []any) error {
```

Comment:
Nit: if we have the map and slice expansions return `(map[string]any, error)` and `([]any, error)` and have `expandValue` return `(any, error)`, we can avoid code duplication by moving the `switch` statement into `expandValue` and reduce the body of the loops in `expandSlice` and `expandMap` to
```
ev, err := expandValue(v)
if err != nil {
	return nil, err
}
yaml<slice | map> = ev
```
If we want to keep all of the current `expandValue` logic in its own function (which makes sense to me), we could rename it to `expandString` and then have `expandValue` only do the switch logic for types, forwarding to `expandMap`, `expandSlice`, or `expandString` as appropriate.